File: python/number_prime/prime.py
class Primes(object):
    num = 3
    primes = [2]
    primes_size = 1

    @classmethod
    def first(self, firsts):
        num = self.num
        primes_size = self.primes_size

        while primes_size < firsts:
            if self._is_prime(num):
                self.primes.append(num)
                primes_size += 1
            num += 2

        self.num = num
        self.primes_size = primes_size

        return Primes.primes[0:firsts]

    @classmethod
    def _is_prime(self, num):
        for divisor in xrange(3, int(num ** 0.5) + 1, 2):
            if not num % divisor:
                return False
        return True

File: php/NumbersPrimes/Prime.php
<?php

class Prime
{
    private static $num = 3;
    private static $primes = [2];
    private static $primesLength = 1;

    public static function first($firsts): array
    {
        $num = self::$num;
        $primesLength = self::$primesLength;

        while ($primesLength < $firsts) {
            if (self::isPrime($num) && !in_array($num, self::$primes)) {
                array_push(self::$primes, $num);
                $primesLength += 1;
            }
            $num += 2;
        }
        self::$primesLength = $primesLength;

        return array_slice(self::$primes, 0, $firsts);
    }

    private static function isPrime($num): bool
    {
        for ($divisor = 3; $divisor <= ceil(sqrt($num)); $divisor += 2) {
            if (!($num % $divisor)) {
                return false;
            }
        }

        return true;
    }
}
